File: processor/src/test/resources/scenarios/cassandracodecs/NoNoArgCassandraCodec.java
package scenarios.cassandracodecs;

import com.datastax.oss.driver.api.core.type.codec.MappingCodec;
import com.datastax.oss.driver.api.core.type.codec.TypeCodecs;
import com.datastax.oss.driver.api.core.type.reflect.GenericType;

public class NoNoArgCassandraCodec extends MappingCodec<String, String> {

    private final String prefix;

    public NoNoArgCassandraCodec(String prefix) {
        super(TypeCodecs.TEXT, GenericType.of(String.class));
        this.prefix = prefix;
    }

    @Override
    protected String innerToOuter(String value) {
        return prefix + value;
    }

    @Override
    protected String outerToInner(String value) {
        return value;
    }
}

File: ksp/src/test/resources/scenarios/cassandracodecs/NoNoArgCassandraCodec.kt
package scenarios.cassandracodecs

import com.datastax.oss.driver.api.core.type.codec.MappingCodec
import com.datastax.oss.driver.api.core.type.codec.TypeCodecs
import com.datastax.oss.driver.api.core.type.reflect.GenericType

class NoNoArgCassandraCodec(private val prefix: String) :
    MappingCodec<String, String>(TypeCodecs.TEXT, GenericType.of(String::class.java)) {
    override fun innerToOuter(value: String?): String? = value?.let { prefix + it }
    override fun outerToInner(value: String?): String? = value
}
